userland/test/CMakeLists.txt

```diff
@@ -4,12 +4,14 @@ foreach(middleware_impl_tmp ${middleware_implementations})
 
   add_executable(test_publisher__${middleware_impl_tmp} "test_publisher.cpp")
   ament_target_dependencies(test_publisher__${middleware_impl_tmp}
+    "rcl_interfaces"
```

Comment:
Agreed, but adding `rcl_interfaces` as a dependency to `rclcpp`'s `package.xml` does not seem to change anything. Is there any other place I should change?